opal/managers.py

```diff
@@ -150,3 +150,71 @@ def serialised_active(self, user, **kw):
         episodes = self.filter(**filters)
         as_dict = self.serialised(user, episodes)
         return as_dict
+
+
+class LookupListQueryset(models.QuerySet):
+    def get_content_type(self):
+        return ContentType.objects.get_for_model(self.model)
+
+    def find_ids_from_synonyms(
+        self, some_strs, contains=False, case_sensitive=False
+    ):
+        """
+        Searches through synonyms and returns
+        the related object ids
+        e.g.
+        if instance top_hat had the synonym topper
+        and some_strs = ['topper']
+        we would return ValuesList(top_hat.id)
+        """
+        from opal.models import Synonym
+        content_type = self.get_content_type()
+        query_arg = self.get_query_arg(contains, case_sensitive)
+        q_objects = []
+        for some_str in some_strs:
+            kwargs = {
+                query_arg: some_str,
+                "content_type": content_type
+            }
+            q_objects.append(Q(**kwargs))
+        return Synonym.objects.filter(
+            reduce(operator.or_, q_objects)
+        ).values_list("object_id", flat=True).distinct()
+
+    def get_query_arg(self, contains, case_sensitive):
+        if contains and not case_sensitive:
+            return "name__icontains"
+        if contains and case_sensitive:
+            return "name__contains"
+        if not contains and not case_sensitive:
+            return "name__iexact"
+        return "name"
+
+    def search(self, some_str, contains=False, case_sensitive=False):
```

Comment:
What's the use case for `case_sensitive` ? I can't think of one...

Meanwhile I can't really think why I wouldn't want contains to be True...

Reply:
For example in search, we offer, `exact` or `contains`, which is the use case of contains. The case insensitive is for completeness.

I will turn contains on by default